### ISSUE TITLE
glslang should report a error : textureOffset not support sampler2DArrayShadow sampler at oes300 version.

### ISSUE THREAD

**Issue by ZhiqianXia** [KhronosGroup/glslang#2596]
Hi ,
   glslang should report an error about **textureOffset** , because **textureOffset** not support **sampler2DArrayShadow**  sampler at oes300.

```
#version 300 es
precision mediump float;
in highp vec4 dEQP_Position;
uniform mediump sampler2DArrayShadow s;
void main()
{
                gl_Position = vec4(textureOffset(s, vec4(0), ivec2(0)));
                gl_Position = dEQP_Position;
}
```
spec link : 
[oes30 spec](https://www.khronos.org/registry/OpenGL/specs/es/3.0/GLSL_ES_Specification_3.00.pdf )  8.8 Texture Lookup Functions

**Comment by greg-lunarg** [comment]
Agreed. It is also not available in oes31 or oes32. Note it is also not available in GLSL 4.20 or earlier.

Are you planning to submit a fix?

**Comment by ZhiqianXia** [comment]
yes, I will give it a try.

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by greg-lunarg** [comment]
Fixed by #2609
